system/src/roller/Roller2D20.mjs

```diff
@@ -532,3 +548,126 @@ export class Roller2D20 {
+		this.rolltype = rolltype;
+	}
+
+	activateListeners(html) {
+		super.activateListeners(html);
+		html.on("change", ".spend-luck", event =>  {
+			let usedluckpoints = 0;
+			if (this.rolltype === 6) {
+				usedluckpoints = Math.ceil(parseInt($(event.currentTarget).val())/ 3);
+			}
+			else {
+				usedluckpoints = parseInt($(event.currentTarget).val());
+			}
+			const nonSpendLuck = html.find(".non-spend-luck");
+			const nonSpendLucknumofdice = parseInt($(nonSpendLuck).val());
+			nonSpendLuck.empty();
+			for (let i = 0; i <= this.numOfDice - usedluckpoints; i++) {
+				const adjustedValue = Math.max(0, i);
+				const selected = i === 0 ? "selected" : "";
+				nonSpendLuck.append(`<option value="${adjustedValue}" ${selected}>${adjustedValue}</option>`);
+				nonSpendLuck[0].selectedIndex = nonSpendLucknumofdice;
+			}
+			const avaliableluck = this.actor.system.luckPoints;
+			if (usedluckpoints > avaliableluck) {
+				const part1 = game.i18n.localize("FALLOUT.UI.Not_Enough");
+				const part2 = game.i18n.localize("FALLOUT.AbilityLuc");
+				const part3 = game.i18n.localize("FALLOUT.TEMPLATES.Points");
+				ui.notifications.warn(`${part1} ${part2} ${part3}`);
+				html.find(".spend-luck")[0].selectedIndex = 0;
+				html.find(".spend-luck").trigger("change");
+
+			}
+		});
+		html.on("change", ".non-spend-luck", event => {
+			const freedicetoreroll = parseInt($(event.currentTarget).val());
+			const SpendLuck = html.find(".spend-luck");
+			const spendLucknumofdice =  parseInt($(SpendLuck).val());
+			SpendLuck.empty(); // Clear existing options
+			for (let i = 0; i <= this.numOfDice - freedicetoreroll; i++) {
+				const adjustedValue = Math.max(0, i);
+				const selected = i === 0 ? "selected" : "";
+				SpendLuck.append(`<option value="${adjustedValue}" ${selected}>${adjustedValue}</option>`);
+				SpendLuck[0].selectedIndex = spendLucknumofdice;
+			}
+		});
+
+	}
+
+	static async spendluck({actordata, numOfDice, rolltype }) {
+		return new Promise((resolve, reject) => {
+			const textnumberofdicetorerollluck = game.i18n.localize("FALLOUT.UI.NUMOFDTOREROLLLUCK");
+			const textnumberofdicetoreroll = game.i18n.localize("FALLOUT.UI.NUMOFDTOREROLL");
+			let optionsHtml = "";
+			for (let i = 0; i <= numOfDice; i++) {
+				if (i===0) {
+					optionsHtml += `<option value="${i}" selected>${i}</option>`;
+				}
+				else {
+					optionsHtml += `<option value="${i}">${i}</option>`;
+				}
+			}
+			const html = `
+			<div class="flexrow fallout-dialog">
+				<div class="flexrow resource" style="padding:5px">
+					<label class="title-label">${textnumberofdicetorerollluck}</label>
+					<select class="spend-luck" name="spend-luck">
+            			${optionsHtml}
+        			</select>
+					<label class="title-label">${textnumberofdicetoreroll}</label>
+					<select class="non-spend-luck" name="non-spend-luc">
+            			${optionsHtml}
+        			</select>
+				</div>
+			</div>`;
+			// eslint-disable-next-line max-len
```

Comment:
I'd like to avoid having html embedded in the code this way, and am gradually migrating any existing instances into template files.

Please could you move all of this to a sensibly named template in the `templates/dialogs` directory and render it using `renderTemplate` (there are examples in the `Dialog2d20.mjs` file if you've not done that before.

Reply:
done